packaging/pulp_repo.py

```diff
@@ -0,0 +1,644 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+from time import sleep
+
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+
+DOCUMENTATION = '''
+---
+module: pulp_repo
+author: "Joe Adams (@sysadmind)"
+short_description: Add or remove Pulp repos from a remote host.
+description:
+  - Add or remove Pulp repos from a remote host.
+version_added: "2.2"
+requirements: []
+options:
+  add_export_distributor:
+    description:
+      - Whether or not to add the export distributor to new C(rpm) repositories.
+    required: false
+    default: false
+  feed_url:
+    description:
+      - Upstream feed URL to receive updates from.
+    required: false
+    default: null
+  force_basic_auth:
+    description:
+      - httplib2, the library used by the M(uri) module only sends
+        authentication information when a webservice responds to an initial
+        request with a 401 status. Since some basic auth services do not
+        properly send a 401, logins will fail. This option forces the sending of
+        the Basic authentication header upon initial request.
+    required: false
+    choices: [ "yes", "no" ]
+    default: "no"
+  importer_ssl_ca_cert:
+    description:
+      - CA certificate string used to validate the feed source SSL certificate.
+    required: false
+    default: null
+  importer_ssl_client_cert:
+    description:
+      - Certificate used as the client certificate when synchronizing the
+        repository. This is used to communicate authentication information to
+        the feed source. The value to this option must be the full path to the
+        certificate. The specified file may be the certificate itself or a
+        single file containing both the certificate and private key.
+    required: false
+    default: null
+  importer_ssl_client_key:
+    description:
+      - Private key to the certificate specified in I(importer_ssl_client_cert),
+        assuming it is not included in the certificate file itself.
+    required: false
+    default: null
+  name:
+    description:
+      - Name of the repo to add or remove. This correlates to repo-id in Pulp.
+    required: true
+  proxy_host:
+    description:
+      - Proxy url setting for the pulp repository importer. This is in the
+        format scheme://host.
+    required: false
+    default: null
+  proxy_port:
+    description:
+      - Proxy port setting for the pulp repository importer.
+    required: false
+    default: null
+  publish_distributor:
+    description:
+      - Distributor to use when state is C(publish). The default is to
+        publish all distributors.
+    required: false
+  pulp_host:
```

Comment:
I would change the postfix to whatever you decide to use with the case of `proxy_host`.